packages/runtime/src/store/previews.ts

```diff
@@ -58,14 +61,12 @@ export class PreviewsStore {
     const previewInfos = previews.map((preview) => {
       const info = new PreviewInfo(preview);
 
-      let previewInfo = this._availablePreviews.get(info.port);
+      let previewInfo = this._availablePreviews.find((availablePreview) => PreviewInfo.equals(info, availablePreview));
 
       if (!previewInfo) {
         previewInfo = info;
 
-        this._availablePreviews.set(previewInfo.port, previewInfo);
-      } else {
-        previewInfo.title = info.title;
```

Comment:
Any ideas why this was here? Removing it doesn't seem to break anything.

Reply:
This is because the preview title can be changed between lessons even if the port hasn't changed.

However now, if we change `_availablePreviews` to be `Map<number, PortInfo>` and instead have all `PreviewInfo` have a reference to an instance in that map, then in here we always re-create the `PreviewInfo` instances.